### PR TITLE
Fixed shade to string method and reduced solution complexity

### LINKED ISSUE #92: Conversion from Shade Panel type to context shade creates IDF string incorrectly
#### Description:

String representation of shade object is not being written correctly, causing simulation to fail.

image

### PR DESCRIPTION
**Pull request by tg359**
### Issues addressed by this PR

Closes #92

Also moves the `ToEnergyPlus` methods into a single place to save on loads of files everywhere.

### Test files

[unnamed.zip](https://github.com/BHoM/EnergyPlus_Toolkit/files/5089603/unnamed.zip)

### CODE REVIEW

File: EnergyPlus_Engine/Convert/ToEnergyPlus.cs [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+                buildingSurface.SunExposure = sunExposure;
+                buildingSurface.WindExposure = windExposure;
+                buildingSurface.Vertices = vertices;
+                buildingSurface.NumberOfVertices = vertexCount;
+                classes.Add(buildingSurface);
+
+                foreach (BHE.Opening o in panel.Openings)
+                {
+                    classes.AddRange(o.ToEnergyPlus(panelName));
+                }
```
Comment (FraserGreenroyd) [inline EnergyPlus_Engine/Convert/ToEnergyPlus.cs:344]:
```suggestion
                foreach (BHE.Opening o in panel.Openings)
                    classes.AddRange(o.ToEnergyPlus(panelName));
```

OR (this one may not work)

```suggestion
                classes.AddRange(panel.Openings.SelectMany(x => x.ToEnergyPlus(panelName)));
```

File: EnergyPlus_Engine/Convert/ToEnergyPlus.cs [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+                    return BuildingSurfaceType.Wall;
+
+                case BHE.PanelType.CurtainWall:
+                    return BuildingSurfaceType.Wall;
+
+                case BHE.PanelType.Roof:
+                    return BuildingSurfaceType.Roof;
+
+                default:
+                    return BuildingSurfaceType.Wall;
```
Comment (FraserGreenroyd) [inline EnergyPlus_Engine/Convert/ToEnergyPlus.cs:376]:
```suggestion
                case BHE.PanelType.Ceiling:
                    return BuildingSurfaceType.Ceiling;
                case BHE.PanelType.Floor:
                    return BuildingSurfaceType.Floor;
                case BHE.PanelType.Wall:
                    return BuildingSurfaceType.Wall;
                case BHE.PanelType.WallInternal:
                    return BuildingSurfaceType.Wall;
                case BHE.PanelType.CurtainWall:
                    return BuildingSurfaceType.Wall;
                case BHE.PanelType.Roof:
                    return BuildingSurfaceType.Roof;
                default:
                    return BuildingSurfaceType.Wall;
```

File: EnergyPlus_Engine/Convert/ToEnergyPlus.cs [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+using BHP = BH.oM.Physical.Constructions;
+
+namespace BH.Engine.EnergyPlus
+{
+    public static partial class Convert
+    {
+        [Description("Convert a boolean to an EnergyPlus friendly Yes or No")]
+        [Input("value", "A True or False value")]
+        [Output("answer", "A Yes or a No")]
+        public static string Boolean(this bool value)
```
[the remark this answers is not in the document]
Comment (tg359) [reply EnergyPlus_Engine/Convert/ToEnergyPlus.cs:46]: Good catch!

**Pull request merged**